### ISSUE TITLE
cannot use dict items method

### ISSUE THREAD

**Issue by ghiodor** [zopefoundation/Products.PythonScripts#35]
This code in a PythonScript works in Zope2:
```
mydict = {'a':'1', 'b':'1'}
values = {}
for key, value in mydict.items():
    values[key] = 'blah' + value
return values
```
In Zope4 I get this error (bottom traceback):
```
     Module script, line 3, in myscript
    <PythonScript at /test/myscript>
    Line 3
    Module RestrictedPython.Guards, line 288, in guarded_iter_unpack_sequence
    Module AccessControl.ZopeGuards, line 264, in __next__
    Module AccessControl.ZopeGuards, line 303, in guard

zExceptions.unauthorized.Unauthorized: You are not allowed to access 'a particular tuple' in this context
```

I see that "items" is in the whitelist of ZopeGuards.py. Can't really follow the AccessControl code...

**Comment by ghiodor** [comment]
using list(mydict.items()) works, no clue why since py3 dict doc loops on mydict.items()

**Comment by jugmac00** [comment]
Maybe it has to do something with
https://stackoverflow.com/questions/10458437/what-is-the-difference-between-dict-items-and-dict-iteritems/10458567#10458567

**Comment by d-maurer** [comment]
ghiodor wrote at 2019-7-7 05:59 -0700:
>using list(mydict.items()) works, no clue why since py3 dict doc loops on mydict.items()

Apparently, the problem is not the access to `items` itself
but to the data structure returned by the `items` call.
In Python 2, `items` returns a list; in Python 3, it returns a kind
of iterator. Apparently, `RestrictedPython` does not yet properly
handle those iterators.

**Comment by ghiodor** [comment]
yeah, that's it, items() returns an iterator (generator?), not a list as documented in python docs. And that's why 2to3 actually casts it to list (luckily for me).

**Comment by icemac** [comment]
This issue is fixed in AccessControl version 4.1. The upcoming 4.1.2 release of Zope will contain it, see zopefoundation/Zope#694.

**Comment by icemac** [comment]
@ghiodor Could you please try if AccessControl 4.1 fixes the issue for you?

**Comment by dataflake** [comment]
@icemac I can confirm that the sample code in the error description works and does not generate an exception in a sandbox using AccessControl 4.1 and Zope master.

**Comment by dataflake** [comment]
This is fixed with AccessControl 4.1, and that gets pulled in automatically when you update to the just released Zope 4.1.2.
